contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -237,14 +237,26 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
         override
         public
     {
+        uint40 nextQueueIndex = getNextQueueIndex();
+        uint40 queueLength = _getQueueLength();
+        if (queueLength - nextQueueIndex < _numQueuedTransactions) {
+            _numQueuedTransactions = queueLength - nextQueueIndex;
+        }
         require(
             _numQueuedTransactions > 0,
             "Must append more than zero transactions."
         );
 
-        uint40 nextQueueIndex = getNextQueueIndex();
         bytes32[] memory leaves = new bytes32[](_numQueuedTransactions);
+
         for (uint256 i = 0; i < _numQueuedTransactions; i++) {
+            if (msg.sender != sequencer) {
+                Lib_OVMCodec.QueueElement memory el = getQueueElement(nextQueueIndex);
+                require(
+                    el.timestamp + forceInclusionPeriodSeconds < block.timestamp,
+                    "Queue transactions cannot be submitted during the sequencer inclusion period."
```

Comment:
maybe `"Queue transactions can only be submitted by sequencer during force inclusion period"`?

Reply:
Can also just be 
```
require(
                   msg.sender == sequencer ||
                    getQueueElement(nextQueueIndex).timestamp + forceInclusionPeriodSeconds < block.timestamp,
                   "Queue transactions can only be submitted by sequencer during force inclusion period."
```
but up to you on style there.

Reply:
Re. the revert message -- I inherited the revert message from the previous implementation so I'm tempted to not change it so I don't change the tests